pkg/scheduler/algorithm/priorities/image_locality.go

```diff
@@ -69,12 +71,21 @@ func calculateScoreFromSize(sumSize int64) int {
 }
 
 // totalImageSize returns the total image size of all the containers that are already on the node.
+// TODO: cover the corner cases of missed matches, e.g,
+// 1. Using Docker as runtime and docker.io/library/test:tag in pod spec, but only test:tag will present in node status
+// 2. Using the implicit registry, i.e., test:tag or library/test:tag in pod spec but only docker.io/library/test:tag
+// in node status; note that if users consistently use one registry format, this should not happen.
 func totalImageSize(nodeInfo *schedulercache.NodeInfo, containers []v1.Container) int64 {
 	var total int64
 
 	imageSizes := nodeInfo.ImageSizes()
 	for _, container := range containers {
-		if size, ok := imageSizes[container.Image]; ok {
+		// apply the default image tag to ensure matching CRI compliant image names
+		name := container.Image
+		if !strings.Contains(name, ":") {
```

Comment:
If this's always the case (cross container runtime), why not update the container name by apiserver, e.g. `setdefault_xxxx`.

Reply:
This check doesn't work if the image is `registry:port/repo/name`.

Reply:
Unfortunately it's not always this case, this patch is a scheduler side workaround. The final fix should happen on kubelet or even runtimes side.

Reply:
> this patch is a scheduler side workaround.

Why we do workaround in scheduler? instead of a final fix?

Reply:
prefer to avoid such kind of image handling cross the components, it's better to let kubelet and apiserver prepare the valid data :)

Reply:
@k82cn Conclusion is not reached in kubelet, actually I will expect a fix in ALL CRI shims as final fix, check my comment here: https://github.com/kubernetes/kubernetes/issues/64413#issuecomment-393772619

Reply:
@smarterclayton Good catch. Theoretically we should call `applyDefaultImageTag` here as kubelet did, while we later found that this call is expensive indeed.

@silveryfu Could you please cover private registry case int he patch as well? e.g. 
```
{Input: "127.0.0.1:80/test/root", Output: "127.0.0.1:80/test/root:latest"},
```

Reply:
@k82cn Moving `applyDefaultImageTag` logic to apiserver is a little bit tricky since this image handling is not container runtime ignorant. Let me discuss with sig-node and feedback here.

Reply:
BTW, please also add a `TODO` for this discussion :)

Reply:
@k82cn Actually, sig-node had decided that CRI should not expose runtime info long time ago, that's why I mentioned the final fix would probably happen on CRI shim side (we reached this agreement in recent discuss with several sig-node folks at least). I will bring this up in next sig-node meeting. 

Now you can see why I suggest to keep a simple fix on scheduler side unless we have idea of how to fix the inconsistency :-)

Reply:
So `.status.nodeInfo.containerRuntimeVersion` is not correct now? But `:last` is only worked for docker :(

At least give doc to highlight that: non-docker runtime has to give a full name when using this feature.

Reply:
> So .status.nodeInfo.containerRuntimeVersion is not correct now? But :last is only worked for docker :(

The info is  correct, but it's not best practice to rely on it. I will bring this to sig-node discussion as well.

`:latest` works for existing runtimes AFAIK, the concern is: it is not guaranteed in the future.

Reply:
>Now you can see why I suggest to keep a simple fix on scheduler side unless we have idea of how to fix the inconsistency :-)

There is also a performance argument for keeping current fix simple. This execution path is really performance sensitive as pods are scheduled and nodes evaluated serially.

We can definitely handle the port number case in the priority function here. But I think a better patch is to let the ImageSizs in scheduler cache to provide the coverage of all variations of the "logically equivalent" image names @resouser.

Reply:
Recall that previous commits in this PR we now rollbacked from did handle the port number case :)

I will add a simple fix for the port number/pvt registry case in the priority function and benchmark it - as it's probably common.

Reply:
I conjecture the ultimate way of doing this is to have the nodes report the *digest* of an image and have the scheduler cache caching a mapping between any logically equivalent image names to the image's digest by consulting the registry itself (*only the registry has the ground truth*).